core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -524,8 +513,29 @@ public synchronized void initialize(URI uri, org.apache.hadoop.conf.Configuratio
     LOG.info("Initializing filesystem with connect details {}",
         Factory.getConnectDetails(alluxioConf));
 
-    mFileSystem = FileSystem.Factory
-        .create(ClientContext.create(subject, alluxioConf).setUriValidationEnabled(!mIsShimFs));
+    // Create FileSystem for accessing Alluxio.
+    // Disable URI validation for non-Alluxio schemes.
+    mFileSystem = FileSystem.Factory.create(ClientContext.create(subject, alluxioConf)
+        .setUriValidationEnabled(uri.getScheme().equals(Constants.SCHEME)));
+  }
+
+  /**
+   * Validates given FS base URI for scheme and authority.
+   *
+   * @param fsUri FS Uri
+   * @throws IOException
+   * @throws IllegalArgumentException
+   */
+  protected void validateFsUri(URI fsUri) throws IOException, IllegalArgumentException {
```

Comment:
maybe move this to `alluxio.hadoop.FileSystem` instead of a default implementation?